cosmoz-default-tree.html

```diff
@@ -23,134 +32,125 @@
 
 	Cosmoz.DefaultTree.prototype = Object.create(Cosmoz.Tree.prototype);
 
-	Cosmoz.DefaultTree.prototype.getNodeByProperty = function (propertyName, propertyValue) {
+	/**
+	 * Returns an Array with all found Nodes.
+	 * Returns an Object with the first found Node.
+	 */
+	Cosmoz.DefaultTree.prototype.getNodeByProperty = function (propertyName, propertyValue, exact, all, nodeList) {
 		if (propertyName === undefined || propertyValue === undefined) {
 			return;
 		}
 
+		// Defaults
+		exact = exact !== undefined ? exact : true;
+		all = all !== undefined ? all : false;
+
 		if (propertyName === 'pathLocator') {
-			return this._getNodeByPathLocator(propertyValue);
+			return this.getNodeByPathLocator(propertyValue);
 		}
 
-		return this._searchAllRoots(propertyName, propertyValue);
+		return this._searchNodeList(propertyName, propertyValue, exact, all, nodeList);
 	};
 
-	Cosmoz.DefaultTree.prototype.getPath = function (node) {
-		if (node === undefined) {
+	/**
+	 * Returns the path String of the found Node(s).
+	 */
+	Cosmoz.DefaultTree.prototype.getPathByProperty = function (propertyName, propertyValue, exact, all, nodeList) {
+		var node = this.getNodeByProperty(propertyName, propertyValue, exact, all, nodeList);
+		if (!node) {
 			return;
 		}
-		return this._getPathByPathLocator(node.pathLocator);
+		return this.getPathString(node.path, propertyName);
```

Comment:
The return statement is not designed for multiple nodes.
I would say redesign the method to only support first-hit single result. (Drop the extra arguments)